lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -65,14 +65,34 @@ sub register {
   $self;
 }
 
-sub _add_default_response {
-  my ($self, $op_spec, $code) = @_;
-  return if $op_spec->{responses}{$code};
-  my $name   = $self->{default_response_name};
-  my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
-  my %schema = ('$ref' => "#/definitions/$name");
+sub _add_default_response_2 {
+  my ($self, $op_spec) = @_;
+  for my $code (@{$self->{default_response_codes}}) {
+    next if $op_spec->{responses}{$code};
+
+    my $name   = $self->{default_response_name};
+    my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
+    my %schema = ('$ref' => "#/definitions/$name");
+    tie %schema, 'JSON::Validator::Ref', $ref, $schema{'$ref'}, $schema{'$ref'};
+    $op_spec->{responses}{$code} = {description => 'Default response.', schema => \%schema};
+  }
+}
+
+sub _add_default_response_3 {
+  my ($self, $op_spec) = @_;
+  my $root_spec    = $self->validator->get(['/']);
+  my $default_name = $self->{default_response_name};
+  my $ref          = $root_spec->{components}{responses}{$default_name} ||= {
+      description => 'default Mojolicious::Plugin::OpenAPI response',
+      content     => {'application/json' => {schema => $self->_default_schema}},
+  };
```

Comment:
Sorry for going back/forth so much, but I do wonder if we can merge these methods again, since they don't seem so different after all:

```
sub _add_default_response {
  my ($self, $op_spec) = @_;
  my $name        = $self->{default_response_name};
  my $schema_data = $self->validator->schema->data;

  my $ref
    = $self->validator->version =~ /^3/
    ? ($schema_data->{components}{responses}{$name} ||= $self->_default_schema_v3)
    : ($schema_data->{definitions}{$name} ||= $self->_default_schema_v2);

  my %schema
    = $self->validator->version =~ /^3/
    ? ('$ref' => "#/components/responses/$name")
    : ('$ref' => "#/definitions/$name");

  tie %schema, 'JSON::Validator::Ref', $ref, $schema{'$ref'}, $schema{'$ref'};

  for my $code (@{$self->{default_response_codes}}) {
    $op_spec->{responses}{$code} ||= \%schema;
  }
}
```

Reply:
Done that.
I used **ge** instead of regex. In my opinion, it expresses intent more clear. And add some fixes and test to your snippet.